sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1143 @@
+     * @brief the CCA procedure finished
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_CCA_DONE.
+     */
+    IEEE802154_RADIO_CONFIRM_CCA,
+} ieee802154_trx_ev_t;
+
+/**
+ * @brief CSMA-CA exponential backoff parameters.
+ */
+typedef struct {
+    uint8_t min;    /**< minimum value of the exponential backoff */
+    uint8_t max;    /**< maximum value of the exponential backoff */
+} ieee802154_csma_be_t;
+
+/**
+ * @brief RX information associated to a frame
+ */
+typedef struct {
+    /**
+     * @brief RSSI of the received packet.
+     *
+     * The RSSI is a measure of the RF power in dBm for the received packet.
+     * The minimum and maximum values are 0 (-174 dBm) and 254 (80 dBm).
+     */
+    uint8_t rssi;
+    uint8_t lqi;    /**< LQI of the received frame */
+} ieee802154_rx_info_t;
+
+/**
+ * @brief TX information of the last transmitted packet.
+ */
+typedef struct {
+    ieee802154_tx_status_t tx_flags; /**< flags of the last transmission */
+    int8_t retrans;                  /**< number of frame retransmissions of the last TX */
+} ieee802154_tx_info_t;
+
+/**
+ * @brief Forwared declaration of the IEEE802.15.4 device descriptor
+ */
+typedef struct ieee802154_dev ieee802154_dev_t;
+
+/**
+ * @brief Prototype of the IEEE802.15.4 device event callback
+ *
+ * @param[in] dev IEEE802.15.4 device descriptor
+ * @param[in] status the status
+ */
+typedef void (*ieee802154_cb_t)(ieee802154_dev_t *dev,
+                                ieee802154_trx_ev_t status);
+
+/**
+ * @brief the IEEE802.15.4 device descriptor
+ */
+struct ieee802154_dev {
+    /**
+     * @brief pointer to the operations of the device
+     */
+    const ieee802154_radio_ops_t *driver;
+    /**
+     * @brief pointer to the context of the device
+     */
+    void *ctx;
+    /**
+     * @brief the event callback of the device
+     */
+    ieee802154_cb_t cb;
+};
+
+/**
+ * @brief IEEE802.15.4 CCA modes
+ */
+typedef enum {
+    IEEE802154_CCA_MODE_ED_THRESHOLD,       /**< CCA using first mode (energy detection) */
+    IEEE802154_CCA_MODE_CARRIER_SENSING,    /**< CCA using second mode (carrier sensing) */
+    IEEE802154_CCA_MODE_ED_THRESH_AND_CS,   /**< CCA using third mode (energy detection AND carrier sensing) */
+    IEEE802154_CCA_MODE_ED_THRESH_OR_CS,    /**< CCA using third mode (energy detection OR carrier sensing) */
+} ieee802154_cca_mode_t;
+
+/**
+ * @brief Auto ACK configuration
+ */
+typedef enum {
+    IEEE802154_AACK_CONF_DISABLED,      /**< Auto ACK is disabled */
+    IEEE802154_AACK_CONF_ENABLED,       /**< Auto ACK is enabled */
+    IEEE802154_AACK_CONF_FRAME_PENDING, /**< Auto ACK is enabled and frame pending bit set in the next ACK packet */
+} ieee802154_aack_conf_t;
+
+/**
+ * @brief Holder of the PHY configuration
+ */
+typedef struct {
+    uint16_t channel;   /**< IEEE802.15.4 channel number */
+    uint8_t page;       /**< IEEE802.15.4 channel page */
+    int16_t pow;        /**< TX power in dBm */
```

Comment:
8 bit should be enough here if this is a whole number.

Reply:
I think so. We can provide a conversion scale or something.

Reply:
Actually, we can simply use a signed 8-bit here:
```
          Min (dBm) Max (dBm)      
AT86RF2xx -17       3         
CC2538    -23       7         
CC2420    -25       0         
mrf24j40  -36.3     0         
kw2xrf    -34       7         
nrf52840  -40       8         
AT86RF215 -20       15        
```

The standard says assumes the TX power goes from somewhere around -32 dBm ("phyTransmitPower = 0 means equal or less than -32 dBm) and 20 dBm (100 mW according to regulations).

Reply:
done